### PR TITLE
Text Processing for text recognition

### PR DESCRIPTION
**Pull request by FacerOfGod**
Goal: Identify ingredients in a text along with their respective quantities and units.

- [x] ~Create Algorithm to detect ingredients~ (very limited and gets very complex solution us AI)
- [x] Implement ChatGBT.
- [x] Easily prompt ChatGBT and receive JSON.
- [x] Add detected ingredients to the inputViewModel.
- [x] Detected ingredients should be visible in the input after exiting the camera.

A next PR is going to be added to correct the `InputViewModel` Input colors and allow the user to add unknown ingredients to the database

### CODE REVIEW

File: app/src/androidTest/java/com/android/feedme/test/camera/CameraTest.kt [hunk 1]
```diff
@@ -125,16 +128,19 @@ class CameraTest : TestCase() {
...
 
       composeTestRule.waitForIdle()
-      composeTestRule.onNodeWithTag("ML Text Box")
-      composeTestRule.onNodeWithTag("ML Text Box inside")
+      composeTestRule.onNodeWithTag("ML Text Box").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("TopBarNavigation").assertIsDisplayed()
     }
   }
+
+  @Test fun MLInputIngredient() {}
```
Comment (JeannedeM) [inline app/src/androidTest/java/com/android/feedme/test/camera/CameraTest.kt:142]: What is this function doing ?

File: app/src/androidTest/java/com/android/feedme/test/camera/TextAnalysisTests.kt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import com.android.feedme.ml.parseResponse
+import com.android.feedme.model.data.IngredientMetaData
+import com.android.feedme.model.data.MeasureUnit
+import io.mockk.*
+import org.json.JSONObject
+import org.junit.Test
+
+class TextAnalysisTests {
+
+  /*@Test
```
Comment (JeannedeM) [inline app/src/androidTest/java/com/android/feedme/test/camera/TextAnalysisTests.kt:14]: We decided to not use this test for the AnalyzedtextforIngredients() function, as it was costing us chat gpt API tokens, for nothing. Keeping this test would have also mean loosing tokens each time the CI would have been run.....

File: app/src/main/java/com/android/feedme/ml/TextRecognition.kt
Comment (JeannedeM) [inline app/src/main/java/com/android/feedme/ml/TextRecognition.kt:1]: Great documentation for these new functions !

File: app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt [hunk 3]
```diff
@@ -43,10 +46,18 @@ class CameraViewModel : ViewModel() {
   // Could be useful later if lastPhoto needs to be accessed outside the view model
   val lastPhoto = _lastPhoto.asStateFlow()
 
+  val listOfIngredient: List<Ingredient> =
+      listOf(Ingredient("Tomate", "DEFAULT_TYPE", "DEFAULT_ID"))
```
Comment (JeannedeM) [inline app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt:50]: Are you sure to keep this default ingredient tomato here ? I thin it was probably meant for testing ?

**Review by JeannedeM — COMMENTED** [review]
Niceee job ! The result is really impressive and it's a big step forward for our application !

**Review by JeannedeM — APPROVED** [review]

**Comment by JeannedeM** [comment]
We decided to merge this PR even though we don't have the required coverage for two reasons : 

- We have enough overall coverage on the app to absorb it.
- We don't wanna spend chat gpt api tokens for testing => we cannot test this part.

File: app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt (hunk 3 above)
Comment (FacerOfGod) [reply app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt]: Yes, it was just for testing :)

**Pull request merged**